src/functional/program_processors/codegens/gtfn/gtfn_module.py

```diff
@@ -43,33 +47,118 @@ class GTFNTranslationStep(
+        #  the respective parameter is used elsewhere, e.g. in a domain construction, as it is
+        #  expected to be scalar there (instead of a SID). We could solve this by:
+        #   1.) Extending the backend to support scalar arguments in a closure (as in embedded
+        #       backend).
+        #   2.) Use SIDs for all arguments and deref when a scalar is required.
+        closure_scalar_parameters = (
+            eve.trees.pre_walk_values(
+                eve.utils.XIterable(program.closures).getattr("inputs").to_list()
+            )
+            .if_isinstance(itir.SymRef)
+            .getattr("id")
+            .map(str)
+            .to_list()
+        )
+        for obj, program_param in zip(args, program.params):
+            # parameter
+            parameter = get_param_description(program_param.id, obj)
+            parameters.append(parameter)
+
+            # argument expression
+            if (
+                isinstance(parameter, interface.ScalarParameter)
+                and parameter.name in closure_scalar_parameters
+            ):
+                # convert into sid
+                parameter_args.append(f"gridtools::stencil::global_parameter({parameter.name})")
+            else:
+                # pass as is
+                parameter_args.append(parameter.name)
+
+        # handle connectivity parameters and arguments
+        for name, connectivity in offset_provider.items():
+            if isinstance(connectivity, Connectivity):
+                if connectivity.index_type not in [np.int32, np.int64]:
+                    raise ValueError(
+                        "Neighbor table indices must be of type `np.int32` or `np.int64`."
+                    )
+
+                # parameter
+                parameters.append(
+                    interface.ConnectivityParameter(
+                        "__conn_" + name.lower(),
+                        connectivity.origin_axis.value,
+                        name,
+                        connectivity.index_type,  # type: ignore[arg-type]
+                    )
+                )
+
+                # connectivity argument expression
+                nbtbl = (
+                    f"gridtools::fn::sid_neighbor_table::as_neighbor_table<"
+                    f"generated::{connectivity.origin_axis.value}_t, "
+                    f"generated::{name}_t, {connectivity.max_neighbors}"
+                    f">(__conn_{name.lower()})"
+                )
+                connectivity_args.append(
+                    f"gridtools::hymap::keys<generated::{name}_t>::make_values({nbtbl})"
+                )  # TODO(havogt): std::forward, type and max_neighbors
+            elif isinstance(connectivity, Dimension):
+                pass
+            else:
+                raise ValueError(
+                    f"Expected offset provider `{name}` to be a `Connectivity` or `Dimension`, "
+                    f"but got {type(connectivity).__name__}."
+                )
```

Comment:
Same here, but even larger codeblock. I don't want to read all these details if I want to get a high-level overview.